packages/mobile/src/components/core/UserGeneratedText.tsx

```diff
@@ -65,20 +86,46 @@ const Link = ({ children, url, ...other }: TextLinkProps & { url: string }) => {
       if (res.data) {
         if (instanceOfTrackResponse(res)) {
           setUnfurledContent(formatTrackName({ track: res.data }))
+          setTo({ screen: 'Track', params: { id: decodeHashId(res.data.id) } })
         } else if (instanceOfPlaylistResponse(res)) {
           setUnfurledContent(formatCollectionName({ collection: res.data[0] }))
+          setTo({
+            screen: 'Collection',
+            params: { id: decodeHashId(res.data[0].id) }
+          })
         } else if (instanceOfUserResponse(res)) {
           setUnfurledContent(formatUserName({ user: res.data }))
+          setTo({
+            screen: 'Profile',
+            params: { id: decodeHashId(res.data.id) }
+          })
         }
       }
     }
   }, [url, shouldUnfurl, unfurledContent, setUnfurledContent])
 
-  return (
-    <TextLink {...other} url={url}>
-      {unfurledContent ?? url}
-    </TextLink>
+  const handlePress = useCallback(
+    (e: GestureResponderEvent) => {
+      if (to) {
+        if ('push' in navigation) {
+          // @ts-ignore
+          navigation.push(to.screen, to.params)
+        } else {
+          // @ts-ignore
+          navigation.navigate(to.screen, to.params)
```

Comment:
wait I'm confused by this, why are we using two different methods?

Reply:
UserGeneratedText may be rendered outside of the context of a Stack navigator. In those cases we want to `navigate` which will generally replace. But when it's rendered in a Stack navigator we want to push